### PR TITLE
[test] Add preformance test of Fin/Vect

### PR DESCRIPTION
**Pull request by fabianhjr**
Closes idris-lang/Idris2#16

### CODE REVIEW

**Comment by fabianhjr** [comment]
Ah messed up  and missed the expected output

**Comment by fabianhjr** [comment]
Empty test (very small N) takes close to 1.2 seconds on my machine. Increased n to a bigger value (500) which takes roughly 2 seconds on my machine.

File: tests/idris2/perf008/run [hunk 1]
```diff
@@ -0,0 +1,3 @@
+$1 --check FinPerf.idr
+
```
Comment (gallais) [inline tests/idris2/perf008/run:2]: Don't you want to run the program too? Otherwise what's the point of the `main`?

Comment (fabianhjr) [reply tests/idris2/perf008/run:2]: I thought the issue was during typechecking, will change the code regardless.

Comment (fabianhjr) [reply tests/idris2/perf008/run:2]: Changed to `--exec main` instead of `--check`. Same performance (roughly 2 seconds for N=500)

**Comment by fabianhjr** [comment]
Fun note, N=1000 goes to 10seconds so there is still performance increase opportunities. :3

Also, the last example on the issue page now behaves better than the first example. uwu

**Pull request merged**